### PR TITLE
[rllib] switch to python logger

### PR DESCRIPTION
**Pull request by ericl**
- add a logger for the rllib module
- agents can configure this logger via a config key
- all rllib components create child loggers of this

- also fix extra newline in sphinx config docs

### CODE REVIEW

File: python/ray/rllib/utils/__init__.py [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+import logging
+
 from ray.rllib.utils.filter_manager import FilterManager
 from ray.rllib.utils.filter import Filter
 from ray.rllib.utils.policy_client import PolicyClient
 from ray.rllib.utils.policy_server import PolicyServer
 
 __all__ = ["Filter", "FilterManager", "PolicyClient", "PolicyServer"]
 
+RLLIB_LOGGER = logging.getLogger(__name__)
```
Comment (richardliaw) [inline python/ray/rllib/utils/__init__.py]: `__name__` evaluates to `ray.rllib.utils` - why not just use `ray.rllib`?

File: python/ray/rllib/__init__.py [hunk 2]
```diff
@@ -16,6 +18,14 @@
 from ray.rllib.evaluation.sample_batch import SampleBatch
 
 
+def _setup_logger():
+    logger = logging.getLogger("ray.rllib")
+    handler = logging.StreamHandler()
+    handler.setFormatter(logging.Formatter("%(levelname)s: %(message)s"))
```
Comment (richardliaw) [inline python/ray/rllib/__init__.py]: I think there was some prior complaint about doing this, but it's probably fine...

File: python/ray/rllib/agents/agent.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ def _setup(self, config):
                                     self._allow_unknown_subkeys)
         self.config = merged_config
         if self.config["log_level"]:
-            RLLIB_LOGGER.setLevel(self.config["log_level"])
+            logging.getLogger("ray.rllib").setLevel(self.config["log_level"])
```
Comment (richardliaw) [inline python/ray/rllib/agents/agent.py]: I guess one odd issue would be when you do two agents and set one to have more verbose logging but it ends up doubling your output...

**Review by richardliaw — APPROVED** [review]

File: python/ray/rllib/__init__.py (hunk 2 above)
Comment (ericl) [reply python/ray/rllib/__init__.py]: This is scoped to just rllib. However, we should consider adding timestamps and info everywhere, it's hard to debug stuff without the basics.

File: python/ray/rllib/agents/a3c/a3c.py [hunk 4]
```diff
@@ -10,6 +10,7 @@
 from ray.rllib.utils import merge_dicts
 from ray.tune.trial import Resources
 
+# yapf: disable
```
Comment (robertnishihara) [inline python/ray/rllib/agents/a3c/a3c.py]: Why this change? It was passing `yapf` before, right?

File: python/ray/rllib/agents/agent.py (hunk 3 above)
Comment (ericl) [reply python/ray/rllib/agents/agent.py]: Yeah but there's not much we can do about that. Two agents in the same process is kind of an antipattern anyways.

File: python/ray/rllib/__init__.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/rllib/__init__.py]: Yeah, I understand; the complaint iirc was that the user should be in full control of the handlers that are provided to the logger (but that might be not be reasonable to expect for RLlib users), so it's fine

**Comment by ericl** [comment]
That's a pretty silly complaint, we should set a good default and, it
really needed, let the user override.

It's not good to say users should configure things themselves.

File: python/ray/rllib/agents/a3c/a3c.py (hunk 4 above)
Comment (ericl) [reply python/ray/rllib/agents/a3c/a3c.py]: Yapf was adding newlines, messing up the doc includes.
